executor/simple.go

```diff
@@ -1048,6 +1050,143 @@ func (e *SimpleExec) executeGrantRole(s *ast.GrantRoleStmt) error {
 	return nil
 }
 
+// Should cover same internal mysql.* tables as DROP USER, so this function is very similar
+func (e *SimpleExec) executeRenameUser(s *ast.RenameUserStmt) error {
+	// Check privileges.
+	// Check `CREATE USER` privilege.
+	// TODO: Also allow UPDATE privilege for the mysql system schema?
+	if !config.GetGlobalConfig().Security.SkipGrantTable {
+		checker := privilege.GetPrivilegeManager(e.ctx)
+		if checker == nil {
+			return errors.New("miss privilege checker")
+		}
+		activeRoles := e.ctx.GetSessionVars().ActiveRoles
+		if !checker.RequestVerification(activeRoles, mysql.SystemDB, "", "", mysql.UpdatePriv) &&
+			!checker.RequestVerification(activeRoles, "", "", "", mysql.CreateUserPriv) {
+			return core.ErrSpecificAccessDenied.GenWithStackByArgs("CREATE USER or UPDATE on mysql.*")
+		}
+	}
```

Comment:
I think you can remove this and rely on planbuilder to add the required permissions. It looks like you're already doing this in the code too.

Reply:
OK, I remember why you need to do the privilege check here. The `visitInfo` does not accept OR conditions :(

This is the same case in the code for `DROP USER`. I would like to suggest we add this equiverlance check in the privilege manager itself - but should run this by @tiancaiamao who is on vacation right now. We can leave as-is and clean up in a followup PR.

Reply:
Since this is a new feature, then let us only support CREATE USER privilege, and only keep the MySQL compatibility 'similar', not bug for bug or support for legacy reasons (like UPDATE priv for mysql.user would be OK instead of CREATE USER). Reason being that RENAME USER is a utility statement and should not be mixed by 'root' privileges to change the version dependent mysql.* tables directly.

Reply:
Test was added and code removed.